lib/src/iterables/partition.dart

```diff
@@ -18,39 +18,29 @@ part of quiver.iterables;
  * Partitions the input iterable into lists of the specified size.
  */
 Iterable<List> partition(Iterable iterable, int size) {
-  return iterable.isEmpty ? [] : new _Partition(iterable, size);
+  if (size <= 0) throw new ArgumentError(size);
```

Comment:
it's necessary because `sync*` will throw the exception on the first call to `moveNext`, and not when you call `partition`